src/main/java/software/amazon/awssdk/crt/http/Http2StreamManager.java

```diff
@@ -0,0 +1,274 @@
+    private static final int DEFAULT_HTTP_PORT = 80;
+    private static final int DEFAULT_HTTPS_PORT = 443;
+    private final static Charset UTF8 = java.nio.charset.StandardCharsets.UTF_8;
+
+    private final int windowSize;
+    private final URI uri;
+    private final int port;
+    private final int maxConnections;
+    private final int idealConcurrentStreamsPerConnection;
+    private final int maxConcurrentStreamsPerConnection;
+    private final CompletableFuture<Void> shutdownComplete = new CompletableFuture<>();
+
+    /**
+     * Factory function for Http2StreamManager instances
+     *
+     * @param options configuration options
+     * @return a new instance of an Http2StreamManager
+     */
+    public static Http2StreamManager create(Http2StreamManagerOptions options) {
+        return new Http2StreamManager(options);
+    }
+
+    private Http2StreamManager(Http2StreamManagerOptions options) {
+        URI uri = options.getUri();
+        if (uri == null) {
+            throw new IllegalArgumentException("URI must not be null");
+        }
+        if (uri.getScheme() == null) {
+            throw new IllegalArgumentException("URI does not have a Scheme");
+        }
+        if (!HTTP.equals(uri.getScheme()) && !HTTPS.equals(uri.getScheme())) {
+            throw new IllegalArgumentException("URI has unknown Scheme");
+        }
+        if (uri.getHost() == null) {
+            throw new IllegalArgumentException("URI does not have a Host name");
+        }
+
+        ClientBootstrap clientBootstrap = options.getClientBootstrap();
+        if (clientBootstrap == null) {
+            throw new IllegalArgumentException("ClientBootstrap must not be null");
+        }
+
+        SocketOptions socketOptions = options.getSocketOptions();
+        if (socketOptions == null) {
+            throw new IllegalArgumentException("SocketOptions must not be null");
+        }
+
+        boolean useTls = HTTPS.equals(uri.getScheme());
+        TlsContext tlsContext = options.getTlsContext();
+        if (useTls && tlsContext == null) {
+            throw new IllegalArgumentException("TlsContext must not be null if https is used");
+        }
+
+        int windowSize = options.getWindowSize();
+        if (windowSize <= 0) {
+            throw new IllegalArgumentException("Window Size must be greater than zero.");
+        }
+
+        int maxConnections = options.getMaxConnections();
+        if (maxConnections <= 0) {
+            throw new IllegalArgumentException("Max Connections must be greater than zero.");
+        }
+        int maxConcurrentStreamsPerConnection = options.getMaxConcurrentStreamsPerConnection();
+        if (maxConcurrentStreamsPerConnection <= 0) {
+            throw new IllegalArgumentException("Max Concurrent Streams Per Connection must be greater than zero.");
+        }
+        int idealConcurrentStreamsPerConnection = options.getIdealConcurrentStreamsPerConnection();
```

Comment:
What does this do?

Reply:
Added documents to `HttpStreamManagerOptions`